### PR TITLE
Fix event buffer connection held open after exception issue

### PR DESCRIPTION
**Pull request by mapingo**

### CODE REVIEW

File: components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java [hunk 1]
```diff
@@ -37,8 +39,11 @@ public InterceptorContext process(final InterceptorContext interceptorContext, f
     }
 
     private Stream<InterceptorContext> streamFromEventBufferFor(final InterceptorContext interceptorContext) {
-        return eventBufferService.currentOrderedEventsWith(interceptorContext.inputEnvelope())
-                .map(jsonEnvelope -> copyWithInput(interceptorContext, jsonEnvelope));
+        final Stream<JsonEnvelope> envelopeStream = eventBufferService.currentOrderedEventsWith(interceptorContext.inputEnvelope());
+        final List<JsonEnvelope> envelopes = envelopeStream.collect(Collectors.toList());
```
Comment (matt-rich) [inline components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java]: static import?

File: components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java [hunk 2]
```diff
@@ -37,8 +39,11 @@ public InterceptorContext process(final InterceptorContext interceptorContext, f
...
+        envelopeStream.close();
```
Comment (matt-rich) [inline components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java]: if an exception occurs is this still skipped?

File: components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java [hunk 3]
```diff
@@ -37,8 +39,11 @@ public InterceptorContext process(final InterceptorContext interceptorContext, f
     }
 
     private Stream<InterceptorContext> streamFromEventBufferFor(final InterceptorContext interceptorContext) {
```
Comment (matt-rich) [inline components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java]: Add changeling entry.

File: components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java (hunk 2 above)
Comment (mapingo) [inline components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java]: This code is run prior to any possible exception being thrown in the Handles methods...so connections will always be closed before continuing with the interceptor chain

Comment (matt-rich) [inline components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java]: ok, changes look good then.

**Pull request merged**